sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileFormatDataWriter.scala

```diff
@@ -79,9 +81,15 @@ abstract class FileFormatDataWriter(
 
   /** Write an iterator of records. */
   def writeWithIterator(iterator: Iterator[InternalRow]): Unit = {
+    var count = 0L
```

Comment:
could we have something like:
```scala
def writeWithMetrics(record: InternalRow): Unit = {
  if (count % CustomMetrics.NUM_ROWS_PER_UPDATE == 0) {
    CustomMetrics.updateMetrics(currentMetricsValues, customMetrics)
  }
  count += 1
  write(record)
}
```
and then update all the places to use this? instead of replicating the code in several places?

Reply:
sounds good. let me update.